doc/OnlineDocs/contributed_packages/mindtpy.rst

```diff
@@ -71,67 +107,179 @@ MindtPy also supports setting options for mip solver and nlp solver.
+
+.. _Bernal et al., 2017: http://www.optimization-online.org/DB_HTML/2017/08/6171.html
+
+A usage example for Feasibility Pump as the initialization strategy is as follows:
+
+.. code::
+
+  >>> pyo.SolverFactory('mindtpy').solve(model,
+  ...                                    strategy='OA',
+  ...                                    init_strategy='FP',
+  ...                                    mip_solver='cplex',
+  ...                                    nlp_solver='ipopt',
+  ...                                    tee=True
+  ...                                    )
+  >>> model.objective.display()
+
+A usage example for Feasibility Pump as the decomposition strategy is as follows:
+
+.. code::
+
+  >>> pyo.SolverFactory('mindtpy').solve(model,
+  ...                                    strategy='FP',
+  ...                                    mip_solver='cplex',
+  ...                                    nlp_solver='ipopt',
+  ...                                    tee=True
+  ...                                    )
   >>> model.objective.display()
 
 
+
+Solve nonconvex MINLPs
+======================
+
+
+Equality Relaxation
+-------------------
+
+To add
+
+.. tmp_duals are the value of the dual variables stored before using deactivate trivial contraints
+.. The values of the duals are computed as follows: (Complementary Slackness)
+
+.. | constraint | c_geq | status at x1 | tmp_dual (violation) |
+.. |------------|-------|--------------|----------------------|
+.. | g(x) <= b  | -1    | g(x1) <= b   | 0                    |
+.. | g(x) <= b  | -1    | g(x1) > b    | g(x1) - b            |
+.. | g(x) >= b  | +1    | g(x1) >= b   | 0                    |
+.. | g(x) >= b  | +1    | g(x1) < b    | b - g(x1)            |
+
+Augmented Penalty
+-----------------
+
+Augmented Penalty refers to the introduction of (non-negative) slack variables on the right hand sides of the just described inequality constraints and the modification of the objective function when assumptions concerning convexity do not hold. (From DICOPT)
+
+
 Global Outer Approximation
----------------------------------------------
+--------------------------
+
+Apart of the decomposition methods for convex MINLP problems [`Kronqvist et al., 2019`_], MindtPy provides an implementation of Global Outer Approximation (GOA) as described in [`Kesavan et al., 2004`_], to provide optimality guaranteed for nonconvex MINLP problems. Here, the validity of the Mixed-integer Linear Programming relaxation of the original problem is guaranteed via the usage of Generalized McCormick envelopes, computed using the package `MC++`_. The NLP subproblems in this case need to be solved to global optimality, which can be achieved through global NLP solvers such as `BARON`_ or `SCIP`_. 
```

Comment:
Apart from the decomposition methods for convex MINLP problems [`Kronqvist et al., 2019`_], MindtPy provides an implementation of Global Outer Approximation (GOA) as described in [`Kesavan et al., 2004`_], to provide optimality guaranteed for nonconvex MINLP problems. Here, the validity of the Mixed-integer Linear Programming relaxation of the original problem is guaranteed via the usage of Generalized McCormick envelopes, computed using the package `MC++`_. The NLP subproblems, in this case, need to be solved to global optimality, which can be achieved through global NLP solvers such as `BARON`_ or `SCIP`_.